### ISSUE TITLE
Map tool

### ISSUE THREAD

**Issue by IchigoDaifuku** [IchigoDaifuku/Protan#4]
Create a map creating tool. Should:
 - Be able to save and load maps
 - Be able to copy tiles
 - Be able to fill areas of the map with a certain type of tile
 - Be able to copy and paste sections of the map
 - Be able to create 'schematics' that can be reused in the future
 - Have a shortcut tile palette
 - Have a way to select tiles and search for specific tiles
 - Have an undo tool and a redo tool

Make the terrain class be able to:
 - Basically be a terrain tool as well
 - Save tiles into a file for future use

The game should also be able to:
 - Load the saved maps
 - Shift the camera when the map is bigger than the screen can show
